nilearn/externals/skimage/random_walker_segmentation.py

```diff
@@ -0,0 +1,517 @@
+        fs = fs.transpose()
+        rhs.append(B * fs)
+    return lap_sparse, rhs
+
+
+def _mask_edges_weights(edges, weights, mask):
+    """
+    Remove edges of the graph connected to masked nodes, as well as
+    corresponding weights of the edges.
+    """
+    mask0 = np.hstack((mask[:, :, :-1].ravel(), mask[:, :-1].ravel(),
+                       mask[:-1].ravel()))
+    mask1 = np.hstack((mask[:, :, 1:].ravel(), mask[:, 1:].ravel(),
+                       mask[1:].ravel()))
+    ind_mask = np.logical_and(mask0, mask1)
+    edges, weights = edges[:, ind_mask], weights[ind_mask]
+    max_node_index = edges.max()
+    # Reassign edges labels to 0, 1, ... edges_number - 1
+    order = np.searchsorted(np.unique(edges.ravel()),
+                            np.arange(max_node_index + 1))
+    edges = order[edges.astype(np.int64)]
+    return edges, weights
+
+
+def _build_laplacian(data, spacing, mask=None, beta=50,
+                     multichannel=False):
+    l_x, l_y, l_z = tuple(data.shape[i] for i in range(3))
+    edges = _make_graph_edges_3d(l_x, l_y, l_z)
+    weights = _compute_weights_3d(data, spacing, beta=beta, eps=1.e-10,
+                                  multichannel=multichannel)
+    if mask is not None:
+        edges, weights = _mask_edges_weights(edges, weights, mask)
+    lap = _make_laplacian_sparse(edges, weights)
+    del edges, weights
+    return lap
+
+
+#----------- Random walker algorithm --------------------------------
+
+
+def random_walker(data, labels, beta=130, mode='bf', tol=1.e-3, copy=True,
+                  multichannel=False, return_full_prob=False, spacing=None):
+    """Random walker algorithm for segmentation from markers.
+
+    Random walker algorithm is implemented for gray-level or multichannel
+    images.
+
+    Parameters
+    ----------
+    data : array_like
+        Image to be segmented in phases. Gray-level `data` can be two- or
+        three-dimensional; multichannel data can be three- or four-
+        dimensional (multichannel=True) with the highest dimension denoting
+        channels. Data spacing is assumed isotropic unless the `spacing`
+        keyword argument is used.
+    labels : array of ints, of same shape as `data` without channels dimension
+        Array of seed markers labeled with different positive integers
+        for different phases. Zero-labeled pixels are unlabeled pixels.
+        Negative labels correspond to inactive pixels that are not taken
+        into account (they are removed from the graph). If labels are not
+        consecutive integers, the labels array will be transformed so that
+        labels are consecutive. In the multichannel case, `labels` should have
+        the same shape as a single channel of `data`, i.e. without the final
+        dimension denoting channels.
+    beta : float
+        Penalization coefficient for the random walker motion
+        (the greater `beta`, the more difficult the diffusion).
```

Comment:
Thanks for the explanation, but it is hard to figure out where the default comes from.